crates/bevy_asset/src/utils.rs

```diff
@@ -0,0 +1,72 @@
+use concurrent_queue::ConcurrentQueue;
+use std::sync::Arc;
+
+pub(crate) struct EventQueue<T>(Arc<ConcurrentQueue<T>>);
```

Comment:
If we make this module private maybe this could just be a type alias?
```suggestion
pub(crate) type EventQueue<T> = Arc<ConcurrentQueue<T>>;
```